### PR TITLE
Fix various measure/layout issues with android flyout

### LINKED ISSUE #5603: Shell view is not working properly.
### Description

In Xamarin we had very rich ShellView support but here in  .NET MAUI ShellView is not working as it was supposed to do. Let's take a look at the following screenshot -
Screenshot_1648432376

1. You can see that only `Shell.FlyoutHeaderTemplate` seems to be working properly but it's not, the `VerticalOptions="Center"` is not working.
2. The background of the entire shell view is black. From where is this color coming from and how to change this? If I set `<Shell BackgroundColor="White" ...>`, It changes the `ActionBar`'s bg color.
3. The `FlyoutItem` is not aligned and positioned well too! 
4. Same goes for the `Shell.FlyoutFooter` too! Why is this at the end of the view! 

The same view in Xamarin.Forms -

Screenshot_1648435265

The MAUI shell view should look like this too!

### Steps to Reproduce

1. Create a New .NET MAUI App with AppShell view.
2. Paste the following code in the **AppShell.xaml** file: 
```xaml
<?xml version="1.0" encoding="utf-8" ?>
<Shell xmlns="http://schemas.microsoft.com/dotnet/2021/maui" xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" x:Class="MyAwesomeApp.AppShell" xmlns:Pages="clr-namespace:MyAwesomeApp" Title="My Awesome App">
    <Shell.FlyoutHeaderTemplate>
        <DataTemplate>
            <Grid BackgroundColor="#1a1a1a" HeightRequest="100">
                <Label Text="MyApp" TextColor="White" FontAttributes="Bold" FontSize="Title" VerticalOptions="Center" HorizontalOptions="Center"/>
            </Grid>
        </DataTemplate>
    </Shell.FlyoutHeaderTemplate>
    <FlyoutItem Title="Home">
        <FlyoutItem.FlyoutIcon>
            <FontImageSource Glyph="&#xE946;" Color="Black" FontFamily="SegoeFluentIcons"></FontImageSource>
        </FlyoutItem.FlyoutIcon>
        <Tab>
            <ShellContent ContentTemplate="{DataTemplate Pages:MainPage}" />
        </Tab>
    </FlyoutItem>
    <FlyoutItem Title="Home">
        <FlyoutItem.FlyoutIcon>
            <FontImageSource Glyph="&#xE946;" Color="Black" FontFamily="SegoeFluentIcons"></FontImageSource>
        </FlyoutItem.FlyoutIcon>
        <Tab>
            <ShellContent ContentTemplate="{DataTemplate Pages:MainPage}" />
        </Tab>
    </FlyoutItem>
    <Shell.FlyoutFooter>
        <StackLayout HorizontalOptions="StartAndExpand">
            <Label Text="My Awesome App" HorizontalOptions="Start" TextColor="GhostWhite" FontAttributes="Bold"/>
        </StackLayout>
    </Shell.FlyoutFooter>
</Shell>
```
3. Run the app.

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 11, API 30

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

Fix various measure/layouts calls to adhere more correctly to .NET MAUI expectations

image

### Issues Fixed

Fixes #5603

### CODE REVIEW

**Review by Redth — APPROVED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellContentView.cs [hunk 1]
```diff
@@ -112,84 +89,19 @@ public void Measure(int widthMeasureSpec, int heightMeasureSpec, int? maxHeightP
...
-
-			//PlatformView.LayoutParameters = layoutParams;
-			//var c = PlatformView.Context;
-			//var l = (int)c.ToPixels(x);
-			//var t = (int)c.ToPixels(y);
-			//var r = (int)c.ToPixels(width) + l;
-			//var b = (int)c.ToPixels(height) + t;
-
-			//PlatformView.Layout(l, t, r, b);
+			//if (_view.Handler is IPlatformViewHandler pvh)
```
Comment (hartez) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellContentView.cs:97]: Comment cleanup

**Review by hartez — CHANGES_REQUESTED** [review]

**Review by hartez — APPROVED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs [hunk 2]
```diff
@@ -534,6 +529,18 @@ void UpdateElevation()
 					ElevationHelper.SetElevation(view, View);
 			}
 
+			protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
+			{
+				var width = widthMeasureSpec.GetSize();
```
Comment (mattleibow) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs:534]: Is this needed? Doesn't seem to be used and also this appears to be an "empty" override? Debugging fun?

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs [hunk 3]
```diff
@@ -534,6 +529,18 @@ void UpdateElevation()
...
+				base.OnMeasure(widthMeasureSpec, heightMeasureSpec);
+			}
+
+			public override LP LayoutParameters
+			{
+				get => base.LayoutParameters;
+				set => base.LayoutParameters = value;
+			}
```
Comment (mattleibow) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs:542]: Same here? Is this used?

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs [hunk 4]
```diff
@@ -259,15 +259,14 @@ protected virtual void UpdateFlyoutFooter()
...
 
 		void UpdateFooterLayout()
 		{
-			if (_footerView != null)
-			{
-				_footerView.LayoutView(0, 0, _rootView.LayoutParameters.Width, MeasureSpecMode.Unspecified.MakeMeasureSpec(0));
-			}
+			_ = _footerView?
+					.Measure(MeasureSpecMode.Exactly
+					.MakeMeasureSpec(_rootView.LayoutParameters.Width), MeasureSpecMode.Unspecified.MakeMeasureSpec(0), null, null);
```
Comment (mattleibow) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs:268]: Why is there a discard `_` here? Can we not just use the method normally?

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs [hunk 5]
```diff
@@ -284,15 +283,12 @@ void UpdateContentLayout()
...
+					cl.BottomMargin = FooterView?.MeasuredHeight ?? 0;
 
-		void UpdateContentBottomMargin()
-		{
-			if (_flyoutContentView?.LayoutParameters is CoordinatorLayout.LayoutParams cl)
-			{
-				cl.BottomMargin = (int)_shellContext.AndroidContext.ToPixels(_footerView?.View.Height ?? 0);
+				_ = _contentView.Measure(
+					MeasureSpecMode.Exactly.MakeMeasureSpec(width),
+					MeasureSpecMode.Exactly.MakeMeasureSpec(height), null, null);
```
Comment (mattleibow) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs:291]: Is the discard needed?

**Review by mattleibow — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs (hunk 4 above)
Comment (PureWeen) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs:268]: It's informational. It says "yea I know this thing returns a size but I'm intentionally ignoring it."

It seems helpful for future folks :-)

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs (hunk 5 above)
Comment (PureWeen) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs:291]: Same answer

**Review by mattleibow — APPROVED** [review]

**Pull request merged**